pyslim/slim_tree_sequence.py

```diff
@@ -490,6 +490,139 @@ def recapitate(self,
+            in units of haploid samples,
+        :param float mutation_rate: A (constant) mutation rate,
+            in units of mutations per nucleotide per unit of time.
+        :param float recombination_rate: A (constant) recombination rate,
+            in units of crossovers per nucleotide per unit of time.
+        :type recombination_map: :class`msprime.RecombinationMap`
+        :param recombination_map: The recombination map, or None,
+            if recombination_rate is specified.
+        :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+        '''
+        if "keep_first_generation" in kwargs:
+            raise ValueError("The keep_first_generation argument is deprecated:"
+                             "the FIRST_GEN flag is no longer used.")
+
+        discrete_msprime = hasattr(msprime, "RateMap")
+
+        if recombination_rate is not None:
+            if recombination_map is not None:
+                raise ValueError("Cannot specify length/recombination_rate along with a recombination map")
+            if discrete_msprime:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0])
+            else:
+                recombination_map = msprime.RecombinationMap(positions = [0.0, self.sequence_length],
+                                                             rates = [recombination_rate, 0.0],
+                                                             num_loci = int(self.sequence_length))
+        if discrete_msprime and ('discrete_genome' not in kwargs):
+            kwargs['discrete_genome'] = True
+
+        slim_indivs = self.individuals_alive_at(0)
+        slim_nodes = []
+        for ind in slim_indivs:
+            slim_nodes.extend(self.individual(ind).nodes)
+        slim_nodes = np.array(slim_nodes)
+    
+        if samples is None:
+            samples=len(slim_nodes)
+        assert(len(slim_nodes) == samples)
+
+        if Ne is None:
+            Ne=1.0
+
+        new_ts = msprime.simulate(
+                            samples,
+                            Ne=Ne,
+                            end_time=slim_time,
+                            mutation_rate=mutation_rate,
+                            recombination_map = recombination_map,
+                            **kwargs)
+        
+        new_nodes = np.where(new_ts.tables.nodes.time == slim_time)[0]
+
+        node_map = np.repeat(tskit.NULL, new_ts.num_nodes)
+        node_map[new_nodes] = np.random.choice(slim_nodes, len(new_nodes), replace=False)
+
+        tables = self.tables
+        self.adjust_tables_time(tables, slim_time)
+        tables.union(new_ts.tables, node_map,
+                add_populations=False,
+                check_shared_equality=False)
+
+        return SlimTreeSequence(tables.tree_sequence())
+
+    def adjust_tables_time(self, tables, time):
```

Comment:
Hm - I think *this* should actually be a `tskit` method of a TableCollection (confusingly), since we've wanted to do it in a few places separately now. In the interest of not doing too many things at once, it might be better to just leave it here as a 'private' method by putting an underscore at the start of the name (but as `pyslim._adjust_tables_time`, not as a method of a tree sequence).

Reply:
That works for now!